eclair-core/src/test/scala/fr/acinq/eclair/channel/states/StateTestsHelperMethods.scala

```diff
@@ -39,7 +38,29 @@ import scala.concurrent.duration._
 /**
  * Created by PM on 23/08/2016.
  */
-trait StateTestsHelperMethods extends TestKitBase with FixtureTestSuite with ParallelTestExecution {
+trait StateTestsBase extends StateTestsHelperMethods with FixtureTestSuite with ParallelTestExecution {
+
+  implicit class ChannelWithTestFeeConf(a: TestFSMRef[State, Data, Channel]) {
+    // @formatter:off
+    def feeEstimator: TestFeeEstimator = a.underlyingActor.nodeParams.onChainFeeConf.feeEstimator.asInstanceOf[TestFeeEstimator]
+    def feeTargets: FeeTargets = a.underlyingActor.nodeParams.onChainFeeConf.feeTargets
+    // @formatter:on
+  }
+
+}
+
+object StateTestsTags {
+  /** If set, channels will use option_static_remotekey. */
+  val StaticRemoteKey = "static_remotekey"
+  /** If set, channels will use option_anchor_outputs. */
+  val AnchorOutputs = "anchor_outputs"
+  /** If set, channels will be public (otherwise we don't announce them by default). */
+  val ChannelsPublic = "channels_public"
+  /** If set, no amount will be pushed when opening a channel (by default we push a small amount). */
+  val NoPushMsat = "no_push_msat"
+}
```

Comment:
Is there a rationale behind leaving some tags as string? E.g. "funding_unconfirmed", "wumbo", ...

Reply:
Yes, I set here only the tags that are used in this file, because they can be used in several different files across the codebase. The other tags you mention are usually only used in their own test suite, so they don't need to be shared.

The rationale is that if the tag is used in a helper function that is called from many test suites, it's probably worth extracting the tag into a constant.